File: key_races.py
# getting all 5 inputs in a single line
s,v1,v2,t1,t2=map(int,input().strip().split())

# the participant receives the text to be entered t seconds later and exactly t milliseconds after he ends typing all the text, the site receives information about it
# so the total delay is 2*t
# likewise, since one character is typed in v secs, the total time to type s characeters is s*v
# so total_time = v*s + 2*t 

t_first = v1*s + 2*t1
t_second = v2*s + 2*t2

if t_first>t_second:
    print("Second")
elif t_second>t_first:
    print("First")
else:
    print("Friendship")

File: using C/Key races.c
#include <stdio.h>

int main()
{
    int s, v1, v2, t1, t2, total_1, total_2;

    scanf("%d%d%d%d%d", &s, &v1, &v2, &t1, &t2);

    total_1 = t1 + s*v1 + t1;
    total_2 = t2 + s*v2 + t2;

    if (total_1 < total_2)
        printf("First\n");
    else if (total_1 > total_2)
        printf("Second\n");
    else if (total_1 == total_2)
        printf("Friendship\n");

    return 0;
}
